tests/fixtures/cassettes/test_update_device_error_handling_device_id.yaml

```diff
@@ -0,0 +1,620 @@
+      - keep-alive
+      Content-Length:
+      - '2'
+      Content-Type:
+      - application/json
+      User-Agent:
+      - python-requests/2.32.3
+      accept:
+      - application/json
+      x-emn-origin-app:
+      - PythonSDK
+      x-emn-origin-app-version:
+      - 0.3.10
+    method: PATCH
+    uri: https://cdn.emnify.net/api/v1/endpoint/-1
+  response:
+    body:
+      string: '{"message":"token contains an invalid number of segments"}'
+    headers:
+      Access-Control-Allow-Headers:
+      - Authorization,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,X-Count-Per-Page,X-Current-Page,X-Sort,X-Total-Count,X-Total-Pages
+      Access-Control-Allow-Methods:
+      - GET, POST, PATCH, PUT, DELETE
+      Access-Control-Allow-Origin:
+      - '*'
+      Access-Control-Expose-Headers:
+      - Authorization,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,X-Count-Per-Page,X-Current-Page,X-Sort,X-Total-Count,X-Total-Pages,Location,X-Count,X-Per-Page,X-Pages,X-Page
+      Access-Control-Max-Age:
+      - '86400'
+      Cache-Control:
+      - max-age=0
+      Connection:
+      - keep-alive
+      Content-Length:
+      - '58'
+      Content-Type:
+      - application/json
+      Date:
+      - Thu, 17 Oct 2024 08:57:53 GMT
+      Server:
+      - CloudFront
+      Strict-Transport-Security:
+      - max-age=31536000
+      Via:
+      - 1.1 3346ad1679ff01fe81e117084323b814.cloudfront.net (CloudFront)
+      X-Amz-Cf-Id:
+      - Xeg9PhpzA6SKLBtyQiiozznqpUb0PnXuROxFZhTboRcenGHUIxv4ZQ==
+      X-Amz-Cf-Pop:
+      - MUC50-P3
+      X-Amzn-Trace-Id:
+      - Root=1-6710d191-3773394244e8806f699b168c
+      X-Cache:
+      - Error from cloudfront
+      X-Content-Type-Options:
+      - nosniff
+      X-Frame-Options:
+      - DENY
+      x-amz-apigw-id:
+      - fyWu1GfqDoEEGGw=
+      x-amzn-ErrorType:
+      - AccessDeniedException
+      x-amzn-RequestId:
+      - 241e8d5f-43b3-4761-b594-f6cfc9a8f407
+    status:
+      code: 401
+      message: Unauthorized
+- request:
+    body: '{}'
+    headers:
+      Accept-Encoding:
+      - gzip, deflate
+      Connection:
+      - keep-alive
+      Content-Length:
+      - '443'
+      Content-Type:
+      - application/json
+      User-Agent:
+      - python-requests/2.32.3
+      accept:
+      - application/json
+      x-emn-origin-app:
+      - PythonSDK
+      x-emn-origin-app-version:
+      - 0.3.10
+    method: POST
+    uri: https://cdn.emnify.net/api/v1/authenticate
+  response:
+    body:
+      string: '{"auth_token":"eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJlc2Mub3JnIjoxNTk2MSwiZXNjLm9yZ05hbWUiOiJSb3Roc2NoZUdtYkgiLCJhcGlfa2V5IjoiIiwiZXNjLmFwcCI6MTMxODgsImVzYy5tYWluT3JnIjoxNTk2MSwiaWF0IjoxNzI5MTU1NDc0LCJleHAiOjE3MjkxNjk4NzQsImF1ZCI6Ii9hcGkvdjEvYXV0aGVudGljYXRpb24iLCJpc3MiOiJlc2EtYXBpIn0.AMzeGvb44JYNtGIBJN3H2rIX8OVgOqTto5BX5eO1tXhev9KTJOgIvaq-TDylLSGdvsg1JqDTNNJKCnMj3JGmlg"}'
```

Comment:
Make sure not to capture these responses even if they are short living.

https://vcrpy.readthedocs.io/en/latest/advanced.html#custom-response-filtering